src/GraphQL/Resolver.hs

```diff
@@ -251,14 +252,14 @@ instance forall ks t f m.
   , HasAnnotatedInputType t
   , Monad m
   ) => BuildFieldResolver m (API.Argument ks t :> f) where
-  buildFieldResolver handler selection@(AST.SelectionField (AST.Field _ _ arguments _ _)) = do
+  buildFieldResolver handler field = do
     argument <- first SchemaError (API.getArgumentDefinition @(API.Argument ks t))
     let argName = getName argument
+    let arguments = getArguments field
```

Comment:
Just realised that this code should delegate even more work to the data structures themselves.

Rather than saying "field, get my thy arguments" and then "arguments, get me the argument named thus", we should instead say "field, pop this argument".

That would mean:
  1. the resolver could never resolve the same value twice
  2. half as much time spent on lookups

So, something like:

```haskell
let argName = getName argument
(field', value) <- case popArgument field (defaultFor argName) of
  Nothing -> throwError (ValueMissing argName)
  Just (field', v) -> do
    v' <- first (InvalidValue argName) (GValue.fromValue @t v)
    pure (field', v)
buildFieldResolver @m @f (handler value) field'
```

Where,

```haskell
popArgument :: Field -> Name -> Maybe Value -> Maybe Value
popArgument field name default =
  case lookup name (getArguments field) <|> default of
    Nothing -> Nothing
    Just value -> Just (deleteArgument field name, value)
  where
    deleteArgument = notImplemented
```